CA-283664: xen: Ignore errors when writing the dm state to xenstore

Ignore errors from xs_write() since it fails if QEMU is running
deprivileged. The xs_write() is kept since it is still required when
using libxl to start a VM.

This patch will not be needed once libxl uses QMP instead of xenstore
during QEMU startup.

diff --git a/hw/xen/xen-common.c b/hw/xen/xen-common.c
index 4056420..d1cbeee 100644
--- a/hw/xen/xen-common.c
+++ b/hw/xen/xen-common.c
@@ -101,10 +101,12 @@ static void xenstore_record_dm_state(struct xs_handle *xs, const char *state)
     }
 
     snprintf(path, sizeof (path), "device-model/%u/state", xen_domid);
-    if (!xs_write(xs, XBT_NULL, path, state, strlen(state))) {
-        fprintf(stderr, "error recording dm state\n");
-        exit(1);
-    }
+    /*
+     * Ignore the return value since xs_write() will fail when running
+     * deprivileged. The failure is not important since in that case it uses
+     * QMP instead of xenstore for startup communication.
+     */
+    xs_write(xs, XBT_NULL, path, state, strlen(state));
 }
 
 
